providers/src/airflow/providers/edge/cli/api_client.py

```diff
@@ -0,0 +1,120 @@
+
+from airflow.configuration import conf
+from airflow.exceptions import AirflowException
+from airflow.providers.edge.worker_api.auth import jwt_signer
+from airflow.providers.edge.worker_api.datamodels import WorkerStateBody
+
+if TYPE_CHECKING:
+    from airflow.providers.edge.models.edge_worker import EdgeWorkerState
+
+logger = logging.getLogger(__name__)
+
+
+class AirflowHttpException(AirflowException):
+    """Raise when there is a problem during an http request on the internal API decorator."""
+
+    def __init__(self, message: str, status_code: HTTPStatus):
+        super().__init__(message)
+        self.status_code = status_code
+
+
+def _is_retryable_exception(exception: BaseException) -> bool:
+    """
+    Evaluate which exception types to retry.
+
+    This is especially demanded for cases where an application gateway or Kubernetes ingress can
+    not find a running instance of a webserver hosting the API (HTTP 502+504) or when the
+    HTTP request fails in general on network level.
+
+    Note that we want to fail on other general errors on the webserver not to send bad requests in an endless loop.
+    """
+    retryable_status_codes = (HTTPStatus.BAD_GATEWAY, HTTPStatus.GATEWAY_TIMEOUT)
+    return (
+        isinstance(exception, AirflowHttpException)
+        and exception.status_code in retryable_status_codes
+        or isinstance(exception, (ConnectionError, NewConnectionError))
+    )
+
+
+@tenacity.retry(
+    stop=tenacity.stop_after_attempt(10),
+    wait=tenacity.wait_exponential(min=1),
+    retry=tenacity.retry_if_exception(_is_retryable_exception),
+    before_sleep=tenacity.before_log(logger, logging.WARNING),
+)
+def _make_generic_request(method: str, rest_path: str, data: str) -> Any:
+    signer = jwt_signer()
+    api_url = conf.get("edge", "api_url")
+    path = urlparse(api_url).path.replace("/rpcapi", "")
+    headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "Authorization": signer.generate_signed_token({"method": str(Path(path, rest_path))}),
+    }
+    api_endpoint = urljoin(api_url, rest_path)
+    response = requests.request(method, url=api_endpoint, data=data, headers=headers)
+    if response.status_code == 204:
+        return None
+    if response.status_code != 200:
+        raise AirflowHttpException(
+            f"Got {response.status_code}:{response.reason} when sending "
+            f"the internal api request: {response.text}",
+            HTTPStatus(response.status_code),
+        )
+    return json.loads(response.content)
+
+
+def worker_register(
+    hostname: str, state: EdgeWorkerState, queues: list[str] | None, sysinfo: dict
+) -> datetime:
+    """Register worker with the Edge API."""
+    result = _make_generic_request(
+        "POST",
+        f"worker/{quote(hostname)}",
+        WorkerStateBody(state=state, jobs_active=0, queues=queues, sysinfo=sysinfo).model_dump_json(),
```

Comment:
```suggestion
        WorkerStateBody(state=state, jobs_active=0, queues=queues, sysinfo=sysinfo).model_dump_json(exclude_unset=True),
```

if you want to ignore sending defaults!